packages/react/src/QuestionnaireForm/QuestionnaireForm.tsx

```diff
@@ -69,14 +69,29 @@ export function QuestionnaireForm(props: QuestionnaireFormProps): JSX.Element |
         Array.isArray(newResponseItems) ? newResponseItems : [newResponseItems]
       );
 
-      const newResponse: QuestionnaireResponse = {
+      const tempResponse: QuestionnaireResponse = {
         resourceType: 'QuestionnaireResponse',
         status: 'in-progress',
         item: mergedItems,
       };
+
+      const updatedItems = evaluateCalculatedExpressions(questionnaire?.item ?? [], tempResponse);
```

Comment:
Once responses are added to QuestionnaireResponse. Then go through all questionnaire.item checking whether they have an expression to be evaluated and it is does use tempResponse to evaluate.